libs/ledger/src/chain/main_chain.cpp

```diff
@@ -1549,15 +1547,15 @@ DigestSet MainChain::DetectDuplicateTransactions(BlockHash const &starting_hash,
   }
 
   DigestSet potential_duplicates{};
-  for (auto const &digest : transactions)
+  for (auto const &tx_layout : transactions)
   {
-
-    std::pair<bool, std::size_t> const result = bloom_filter_->Match(digest);
+    std::pair<bool, std::size_t> const result =
```

Comment:
Too huge a return type, would `auto` be better here?

Reply:
Done.